Tests/FluentAssertions.Specs/Primitives/ReferenceTypeAssertionsSpecs.cs

```diff
@@ -407,6 +408,30 @@ public void When_an_assertion_on_two_unknown_objects_fails_it_should_report_the_
     }
 
     #endregion
+
+    public class Miscellaneous
+    {
+        [Fact]
+        public void Should_throw_a_helpful_error_when_accidentally_using_equals()
+        {
+            // Arrange
+            var subject = new ReferenceTypeAssertionsDummy(null);
+
+            // Act
+            Action action = () => subject.Equals(subject);
+
+            // Assert
+            action.Should().Throw<NotSupportedException>().WithMessage("Equals is not part of Fluent Assertions. Did you mean Be() or BeSameAs() instead?");
+        }
+
+        public class ReferenceTypeAssertionsDummy : ReferenceTypeAssertions<object, ReferenceTypeAssertionsDummy>
```

Comment:
Most but not all classes inheriting from `ReferenceTypeAssertions` override `Equals()`.  I figure using this dummy class here is good enough to test this.  An alternative is picking one of the other inheriting classes such as `AssemblyAssertions` but that feels like it could change later and the test could begin asserting on the wrong thing without someone realizing.